plugins/samples/domain_rewrite/plugin.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START serviceextensions_plugin_domain_rewrite]
+use lol_html::*;
+use proxy_wasm::traits::*;
+use proxy_wasm::types::*;
+use std::cell::RefCell;
+use std::error::Error;
+use std::rc::Rc;
+
+proxy_wasm::main! {{
+    proxy_wasm::set_log_level(LogLevel::Trace);
+    proxy_wasm::set_http_context(|_, _| -> Box<dyn HttpContext> { MyHttpContext::new()});
+}}
+
+struct MyHttpContext<'a> {
+    // Some member variables need to be wrapped in Rc<Refcell<T>> so that we can
+    // modify the variable in multiple places as well as share ownership. By
+    // using Rc, we are commiting to not modfying the variable in multiple places
```

Comment:
I didn't follow this comment. Might be worth breaking it down:
- Rc is a refcounted pointer. Why do we need it? I see output.clone() and output.borrow_mut(), but it feels like the clone (refcounting) could be avoided. Maybe you could initialize MyHttpContext.output first, then set MyHttpContext.rewriter. 
- Refcell is a runtime mutable-reference tracker (?). I don't see it used in the [example documentation](https://docs.rs/lol_html/latest/lol_html/struct.HtmlRewriter.html#example), so I'm not clear why we need it here. There shouldn't be any threading issues because we'll never run callbacks for a single stream with any parallelism.

Reply:
OK I think I understand the need for Rc:
https://users.rust-lang.org/t/how-to-create-a-struct-with-fields-which-reference-each-other/78260

And that leads to the need for Refcell:
https://stackoverflow.com/questions/52994205/what-is-the-standard-way-to-call-a-mutable-method-in-a-rc-wrapped-object

But this is pretty crazy! Ultimately these are a bunch of mutable fields in a mutable struct. Let's ask around.

Reply:
Asked some Rust folks. They concur with the type used. Related code review feedback:

Box<dyn FnMut(..)> is unnecessary: you can create your own OutputSink type that writes to the Vec without the extra allocation. Relatedly, a struct ExtendSink<T>(pub T) that impl<T: iter::Extend<u8>> OutputSink for ExtendSink<T> would be a generally useful helper struct: you'd use ExtendSink<Vec<u8>> here.